### PR TITLE
Create event agent benchmark

### PR DESCRIPTION
**Pull request by AymericDu**
New benchmark for the event agent.

It can manage the events:
- `storage.chunk.new`
- `storage.chunk.deleted`
- `storage.container.new`
- `storage.container.state`
- `storage.container.deleted`
- `storage.content.deleted`

For example:
```shell
$ oio-event-benchmark OPENIO CHUNK_NEW -q
Linking fake rawx address with the fake service address...
Starting the fake service...
Sending 100 fake events
100 events sent (errors: 0) in 0.095444 seconds, 1047.734797 events/sec
Sending 1047 fake events
1047 events sent (errors: 0) in 0.724469 seconds, 1445.196413 events/sec
Sending 1445 fake events
^CStopping the fake service...
1445 events sent (errors: 0) in 1.025482 seconds, 1409.093480 events/sec
Cleaning...
```

### CODE REVIEW

File: core/dir.c [hunk 1]
```diff
@@ -74,6 +74,21 @@ oio_directory__set_properties(struct oio_directory_s *self,
 	DIR_CALL(self, set_prop)(self, url, values);
 }
 
+GError *
+oio_directory__force (struct oio_directory_s *self,
```
Comment (fvennetier) [inline core/dir.c]: Please don't put a space between a function name and the open parenthesis. This is reserved for conditions (`if`, `for`, `while`).

File: core/dir.c [hunk 2]
```diff
@@ -100,6 +115,13 @@ static GError * _dir_proxy_get_prop(struct oio_directory_s *self,
 static GError * _dir_proxy_set_prop(struct oio_directory_s *self,
 		const struct oio_url_s *url, const char * const *values);
 
+static GError * _dir_proxy_force (struct oio_directory_s *self,
+			const struct oio_url_s *url, const char *srvtype,
+			const char* const *values);
```
Comment (fvennetier) [inline core/dir.c]: Two tabulations are enough.

File: core/dir.c [hunk 3]
```diff
@@ -307,3 +331,37 @@ _dir_proxy_set_prop(struct oio_directory_s *self, const struct oio_url_s *url,
 	oio_url_pclean(&u);
 	return err;
 }
+
+static GError *
+_dir_proxy_force (struct oio_directory_s *self,	const struct oio_url_s *url,
```
Comment (fvennetier) [inline core/dir.c]: You used a tabulation instead of a space.

File: core/proxy.c [hunk 4]
```diff
@@ -838,6 +838,45 @@ oio_proxy_call_reference_set_properties(CURL *h, struct oio_url_s *u,
 	return err;
 }
 
+GError *
+oio_proxy_call_reference_force (CURL *h, struct oio_url_s *u,
+		const char *t, const char* const *values)
+{
+	GString *http_url = _curl_reference_url (u, "force");
+	if (!http_url) return BADNS();
```
Comment (fvennetier) [inline core/proxy.c]: Please don't do one-liner `if` statements.

File: tools/benchmark_event/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,21 @@
+add_definitions(-DG_LOG_DOMAIN="oio.tools.bench")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+
+include_directories(BEFORE . ../..
+		${CMAKE_CURRENT_BINARY_DIR})
+
+add_executable(
+	bench_event_agent
```
Comment (fvennetier) [inline tools/benchmark_event/CMakeLists.txt]: All our executable files use dashes as word separator. The `bean_prefix` macro may help you. The executable file should be something like `oio-event-benchmark`. The directory has to be renamed too.

File: tools/benchmark_event/bench_conf.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#ifndef OIO_SDS__tools__benchmark_event__bench_conf_h
+#define OIO_SDS__tools__benchmark_event__bench_conf_h
+
+#define NAME_SPACE "OPENIO"
```
Comment (fvennetier) [inline tools/benchmark_event/bench_conf.h]: We have always spelled `NAMESPACE` as a single word.

File: tools/benchmark_event/bench_event_agent.c [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+	struct oio_cs_registration_s reg = {
+		.id = FAKE_SERVICE_ADDRESS,
+		.url = FAKE_SERVICE_ADDRESS,
+		.kv_tags = kv
+	};
+
+	GError *err = oio_cs_client__lock_service(cs, NAME_SRVTYPE_ACCOUNT, &reg,
+											  SCORE_MAX);
+	if (err) {
+		GRID_ERROR("Failed to load the lock service: %d %s", err->code,
```
Comment (fvennetier) [inline tools/benchmark_event/bench_event_agent.c]: "Failed to load the lock service"?!

File: tools/benchmark_event/bench_event_agent.c [hunk 8]
```diff
@@ -0,0 +1,358 @@
...
+
+	return cli_options;
+}
+
+static const char *
+grid_main_get_usage (void)
+{
+	return "CHUNK_NEW/CHUNK_DELETED"
+			"/CONTAINER_NEW/CONTAINER_STATE/CONTAINER_DELETED"
+			"/CONTENT_DELETED";
```
Comment (fvennetier) [inline tools/benchmark_event/bench_event_agent.c]: The usual separator for optional command line arguments is `|`.

File: tools/oio-event-benchmark/event_benchmark.c [hunk 9]
```diff
@@ -0,0 +1,353 @@
...
+	if (err) {
+		GRID_ERROR("Failed to list account services: (%d) %s", err->code,
+				err->message);
+		g_clear_error(&err);
+
+		return NULL;
+	}
+
+	struct oio_cs_registration_s * account_reg = services->data;
+	g_slist_free(services);
```
Comment (fvennetier) [inline tools/oio-event-benchmark/event_benchmark.c]: What about the other account services? There can be several ones.

File: tools/oio-event-benchmark/fake_service.c [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+extern gint errors;
+extern gint64 reception_time;
+extern gdouble speed;
+
+gboolean fake_service_ready = FALSE;
+
+static struct path_parser_s *path_parser = NULL;
+static struct network_server_s *server = NULL;
+
+GMutex mutex;
```
Comment (fvennetier) [inline tools/oio-event-benchmark/fake_service.c]: You should make it static.

File: tools/oio-event-benchmark/event_benchmark.c [hunk 11]
```diff
@@ -0,0 +1,353 @@
...
+// Main callbacks
+
+static struct grid_main_option_s *
+grid_main_get_options(void)
+{
+	static struct grid_main_option_s cli_options[] = {
+		{
+			"Rounds", OT_UINT, {.i = &rounds},
+			"Number of rounds for a test"
+		},
```
Comment (AymericDu) [inline tools/oio-event-benchmark/event_benchmark.c]: `Rounds` indicates the number of experiments to average. But if you want to do the same test indefinitely, you can add `Increment=0` in the options.

File: tools/event-benchmark/event_benchmark.c [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+
+static void
+grid_main_specific_stop(void)
+{
+}
+
+static void
+grid_main_specific_fini(void)
+{
+	printf("Stoping the fake service...\n");
```
Comment (fvennetier) [inline tools/event-benchmark/event_benchmark.c]: "Stopping" with two 'p'.

**Pull request merged**